CONTRIBUTING.md

```diff
@@ -0,0 +1,234 @@
+
+# Stacks Project Guidelines
+
+- [Introduction](#introduction)
+- [Get Started](#get-started)
+  - [System Requirements](#system-requirements)
+  - [Setup](#setup)
+- [Naming Conventions and Coding Style](#naming-conventions-and-coding-style)
+- [Creating a New Plugin](#creating-a-new-plugin)
+  - [Generator Analysis](#generator-analysis)
+  - [Development process](#development-process)
+- [Testing](#testing)
+  - [Unit Testing](#unit-testing)
+  - [End to end testing](#end-to-end-testing)
+  - [Testing packages locally](#testing-packages-locally)
+- [Contributing Changes](#contributing-changes)
+- [Releasing packages and publishing to NPM](#releasing-packages-and-publishing-to-npm)
+
+## Introduction
+
+Welcome to the Stacks project! This document provides guidelines to help you get started with the Stacks workspace and develop new plugins or contribute to existing ones. It covers setup instructions, development practices, testing, and contribution guidelines.
+
+## Get Started
+
+### System Requirements
+
+To set up the Stacks workspace, ensure that you have the following installed on your system:
+
+- npm
+
+- Node.js version 18 or above
+
+The following are recommended plugins for visual studio code:
+
+- [Nx Console](https://marketplace.visualstudio.com/items?itemName=nrwl.angular-console)
+
+- [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+
+- [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)
+
+- [Jest Runner](https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner)
+
+### Setup
+
+To start working in the `stacks-nx-plugins` run the following command:
+
+```bash
+
+npm run prepare
+
+```
+
+This will set up [Husky](https://typicode.github.io/husky/) to manage commit quality which enforces naming and documentation conventions upon commits through [commitizen](https://commitizen-tools.github.io/commitizen/).
+
+## Naming Conventions and Coding Style
+
+When working with the Stacks project, adhere to the following naming conventions and coding style guidelines:
+
+- Ensure husky has been configured to enforce commit standards

```

Comment:
how about just adding some brief info (with an example) of what the commit standards are?

Reply:
Updated, please review